lib/src/pubspec.dart

```diff
@@ -517,8 +529,11 @@ class Pubspec {
             devDependencies == null ? null : devDependencies.toList(),
         _dependencyOverrides =
             dependencyOverrides == null ? null : dependencyOverrides.toList(),
-        _dartSdkConstraint = dartSdkConstraint ?? _defaultSdkConstraint,
+        _dartSdkConstraint = dartSdkConstraint ?? includeDefaultSdkConstraint
+            ? _defaultUpperBoundSdkConstraint
+            : VersionConstraint.any,
         _flutterSdkConstraint = flutterSdkConstraint,
+        _includeDefaultSdkConstraint = false,
```

Comment:
I see the ctor arg – then this setting the field ... not consistent?

Reply:
I can see how it would seem that way :), but I am just shortcutting here and setting the `_dartSdkConstraint` directly. I could leave this as `true` but it wouldn't do anything.

Reply:
Oh wait nvm you are right because you can pass in a special dartSdkConstraint so ya I should respect this setting as well blergs.